python/lsst/analysis/ap/analysisPlots/hitsDiaPlot.py

```diff
@@ -0,0 +1,92 @@
+# This file is part of analysis_ap.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ("HitsDiaPlot",)
+
+from lsst.analysis.tools.actions.plot.diaSkyPlot import DiaSkyPanel, DiaSkyPlot
+from lsst.analysis.tools.actions.vector import DownselectVector, RangeSelector
+from lsst.analysis.tools.interfaces import AnalysisPlot
+
+
+class HitsDiaPlot(AnalysisPlot):
+    """Specialized plotter for the DECam HiTS dataset
+    for plotting RA/Dec of DiaSources.
+
+    It makes two panels that make the HiTS fields look pretty.
+    """
+
+    def setDefaults(self):
+        super().setDefaults()
+
+        # Filter data into `ra1s` and `dec1s` for one panel,
+        # and `ra2s` and `dec2s` for the other`
+        self.process.buildActions.ra1s = DownselectVector(
+            vectorKey="ra", selector=RangeSelector(column="dec", maximum=-2)
+        )
+        self.process.buildActions.dec1s = DownselectVector(
+            vectorKey="dec", selector=RangeSelector(column="dec", maximum=-2)
```

Comment:
Also since everything is selected the same way, consider using the selector in prep instead, that way it will operate once, and the results from the "prep" stage will be already useful to someone.

It also occurs to me, if you use the selector in prep, you likely don't even need any process actions, because you could just set the column names directly in the produce stage.

Again, this is not a wrong way to do things, there is so much flexibility to enable people to do anything that often there is no obvious "right" way, so I want to point out other ways so others can decide what is best practice over time.

Reply:
I got burned early on by trying to skip one of the "p" steps altogether and learning I couldn't, so this hadn't occurred to me. Your way is probably neater, but I'm reluctant to mess with this because it is working 😅